dvc/repo/experiments/executor/local.py

```diff
@@ -213,3 +215,58 @@ def cleanup(self, infofile: str):
             checkpoint = self.scm.get_ref(EXEC_CHECKPOINT)
             if checkpoint and checkpoint != self._orig_checkpoint:
                 self.scm.set_ref(EXEC_APPLY, checkpoint)
+
+    @classmethod
+    def save(
+        cls,
+        info: "ExecutorInfo",
+        is_checkpoint: bool = False,
+        force: bool = False,
+    ) -> ExecutorResult:
+        from dvc.repo import Repo
+
+        exp_hash: Optional[str] = None
+        exp_ref: Optional[ExpRefInfo] = None
+
+        dvc = Repo(os.path.join(info.root_dir, info.dvc_dir))
+        old_cwd = os.getcwd()
+        if info.wdir:
+            os.chdir(os.path.join(dvc.scm.root_dir, info.wdir))
+        else:
+            os.chdir(dvc.root_dir)
+
+        try:
+            stages = dvc.commit([], force=force)
+            exp_hash = cls.hash_exp(stages)
```

Comment:
Another comment from the DVCLive scenario (we can discuss if we want to address those in a follow-up) cc @dberenbaum 

The current way of commuting exp_hash is problematic in the "no pipeline" because there are no stages in the `dvc.yaml` (or even there is no `dvc.yaml`), so multiple calls to `experiments.save` will result in a conflict (unless a custom name is passed).

In a repo with no `dvc.yaml`, the following fails with `dvc.repo.experiments.exceptions.ExperimentExistsError`:

```py
from dvc.repo import Repo

repo = Repo()

for i in range(3):
    with open("params.yaml", "w") as f:
        f.write(f"foo: {i}")
    repo.experiments.save()
```

But this works as expected:


```py
from dvc.repo import Repo

repo = Repo()

for i in range(3):
    with open("params.yaml", "w") as f:
        f.write(f"foo: {i}")
    repo.experiments.save(name=str(i))
```

```console
$ dvc exp show
  Experiment        Created    foo  
 ────────────────────────────────── 
  workspace         -          2    
  master            10:28 AM   2    
  ├── 71d48a5 [2]   10:37 AM   2    
  ├── f87cc9e [1]   10:37 AM   1    
  └── 7adc787 [0]   10:37 AM   0    
 ────────────────────────────────── 
```

Reply:
I assume it's more about `dvc.lock` than `dvc.yaml`?

I guess we could have DVCLive auto-generate names for now (related to https://github.com/iterative/vscode-dvc/issues/2457) if it's easier than revisiting how exp names are assigned, but it might be better to track changes across the entire repo instead of only in `dvc.lock`.